### PR TITLE
feat: the merge

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds "the merge" - auto display the best quote for the user (flashmint or 0x)
* Adds swap paths with the format: `/swap/usdc/mvi`
* Adds new `useBalance` hooks, removes old `BalancesProvider`
* Introduces a new `useTrade` hook which accepts a more abstract transaction request `QuoteTransaction` (to work with 0x and flashmint)
* Update `useSimulateQuote` to work with `QuoteTransaction`
* Adds using NAV price for all index tokens instead of index token prices from CoinGecko
* Adds new token utils: `getDefaultIndex`, `getTokenBySymbol`, `isIndexToken`
* Updates app layout to a more centralized layout (w/ header and footer)
* Updates trade component header to now be just a title instead of navigation since flashmint|0x are merged)
* Removes lots of now obsolete code
* Updates a few dependencies (arcx analytics, dotenv, rainbowkit)
* Includes wagmi update that removes ledger https://github.com/wevm/wagmi/releases/tag/wagmi%401.4.12

## Known Issues/Working on fixing
- [x] when redeeming there can be display issues with the USD amount

## Swap Paths / Path resolving
* For the details of how paths are resovled, check `PathResolver`.
* If the token pair is available, it will load as entered
* If not available or the path is just `/swap` or something wrong, a default will be displayed (the current top listed index token + ETH)

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
-  "extends": "next/core-web-vitals",
+  "extends": ["next/core-web-vitals", "plugin:@typescript-eslint/recommended"],
   "rules": {
-    "no-constant-binary-expression": "error"
+    "no-constant-binary-expression": "error",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-unused-vars": "off"
```
Comment (janndriessen) [inline .eslintrc.json:6]: I'll fix those `off` rules over time and turn it back on later. Especially `no-unused-vars` should be good to go before the release.

File: .env.default [hunk 2]
```diff
@@ -2,7 +2,7 @@ NEXT_PUBLIC_ARCX_ANALYTICS_API_KEY=
 NEXT_PUBLIC_ALCHEMY_ID=
 NEXT_PUBLIC_INDEX_COOP_API=
 NEXT_PUBLIC_IP_LOOKUP_KEY=
-NEXT_PUBLIC_GOOGLE_TAG_MANAGER_CONTAINER_ID=GTM-WNPZF4R
+NEXT_PUBLIC_GOOGLE_TAG_MANAGER_CONTAINER_ID=
```
Comment (janndriessen) [inline .env.default:5]: To avoid the next forkers using this key ;)

File: .eslintrc.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
-  "extends": "next/core-web-vitals",
+  "extends": ["next/core-web-vitals", "plugin:@typescript-eslint/recommended"],
   "rules": {
-    "no-constant-binary-expression": "error"
+    "no-constant-binary-expression": "error",
+    "@typescript-eslint/no-explicit-any": "off"
```
Comment (janndriessen) [inline .eslintrc.json:5]: `plugin:@typescript-eslint/recommended` seems like a good addition.

`@typescript-eslint/no-explicit-any` will be fixed step by step in ongoing development.

File: package.json [hunk 4]
```diff
@@ -37,7 +37,7 @@
     "react-dom": "^18.2.0",
     "use-debounce": "^10.0.0",
     "viem": "^1.10.8",
-    "wagmi": "1.4.12"
+    "wagmi": "^1.4.12"
```
Comment (0xonramp) [inline package.json:40]: Thoughts on pinning across our repos? Good points raised here: https://docs.renovatebot.com/dependency-pinning/

File: src/app/swap/[[...path]]/path-resolver.ts [hunk 5]
```diff
@@ -16,46 +21,45 @@ export class PathResolver {
...
+    let inputToken = this.resolveToken(symbols.inputToken ?? ETH.symbol)
+    let outputToken = this.resolveToken(
+      symbols.outputToken ?? defaultIndex.symbol
+    )
+    const inputTokenIsIndex = indicesTokenList.some(
+      (token) => token.symbol === inputToken.symbol
+    )
+    const outputTokenIsIndex = indicesTokenList.some(
+      (token) => token.symbol === outputToken.symbol
+    )
```
Comment (0xonramp) [inline src/app/swap/[[...path]]/path-resolver.ts:43]: Do we need to worry about casing at all?

File: src/components/header/connect.tsx [hunk 6]
```diff
@@ -1,22 +1,28 @@
...
 
-import { useNetwork } from '@/lib/hooks/useNetwork'
+import { useLedgerStatus } from '@/lib/hooks/use-ledger-status'
+import { useNetwork } from '@/lib/hooks/use-network'
 import { logConnect } from '@/lib/utils/api/analytics'
 
 export const Connect = () => {
   const { address } = useAccount()
   const { chainId } = useNetwork()
+  const { isRunningInLedgerLive } = useLedgerStatus()
```
Comment (0xonramp) [inline src/components/header/connect.tsx:16]: Is this used at all, or can we keep it out of the code for now?

File: src/components/swap/components/transaction-review/index.tsx [hunk 7]
```diff
@@ -190,15 +167,46 @@ const ContractSection = ({
...
+  /**
+   * Returns boolean indicating success or null for config error
+   */
+  const makeTrade = async (override: boolean) => {
+    if (!quote) return null
+    try {
+      await executeTrade(quote, override)
+      return true
+    } catch {
+      return false
```
Comment (0xonramp) [inline src/components/swap/components/transaction-review/index.tsx:193]: Make sense to add any logging here, or is it automatically displayed if something fails?

File: src/app/swap/[[...path]]/path-resolver.ts (hunk 5 above)
Comment (janndriessen) [reply src/app/swap/[[...path]]/path-resolver.ts:43]: This is basically handled in `resolveToken` which returns objects we control, so symbols should always have matching formats. https://github.com/IndexCoop/index-app/blob/4d0e35051d5aa66e9932ea653bc8b2a65966634a/src/app/swap/%5B%5B...path%5D%5D/path-resolver.ts#L84

File: src/components/header/connect.tsx (hunk 6 above)
Comment (janndriessen) [reply src/components/header/connect.tsx:16]: Will remove. 👍  Missed this during rebasing.

File: package.json (hunk 4 above)
Comment (janndriessen) [reply package.json:40]: 💯 was gonna discuss the same.

File: src/components/swap/components/transaction-review/index.tsx (hunk 7 above)
Comment (janndriessen) [reply src/components/swap/components/transaction-review/index.tsx:193]: I'll add logging for this with upcoming task of adding a new error logging provider (if not logged automatically).

**Pull request merged**